packages/components/nodes/chatmodels/ChatGooglePaLM/ChatGooglePaLM.ts

```diff
@@ -0,0 +1,140 @@
+        this.description = 'Wrapper around Google MakerSuite PaLM large language models using the Chat endpoint'
+        this.baseClasses = [this.type, ...getBaseClasses(ChatGooglePaLM)]
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: ['googleMakerSuite']
+        }
+        this.inputs = [
+            {
+                label: 'Model Name',
+                name: 'modelName',
+                type: 'options',
+                options: [
+                    {
+                        label: 'models/chat-bison-001',
+                        name: 'models/chat-bison-001'
+                    }
+                ],
+                default: 'models/chat-bison-001',
+                optional: true
+            },
+            {
+                label: 'Temperature',
+                name: 'temperature',
+                type: 'number',
+                step: 0.1,
+                default: 0.7,
+                optional: true,
+                description:
+                    'Controls the randomness of the output.\n' +
+                    'Values can range from [0.0,1.0], inclusive. A value closer to 1.0 ' +
+                    'will produce responses that are more varied and creative, while ' +
+                    'a value closer to 0.0 will typically result in more straightforward ' +
+                    'responses from the model.'
+            },
+            {
+                label: 'Top Probability',
+                name: 'topP',
+                type: 'number',
+                step: 0.1,
+                optional: true,
+                additionalParams: true,
+                description:
+                    'Top-p changes how the model selects tokens for output.\n' +
+                    'Tokens are selected from most probable to least until ' +
+                    'the sum of their probabilities equals the top-p value.\n' +
+                    'For example, if tokens A, B, and C have a probability of .3, .2, and .1 ' +
+                    'and the top-p value is .5, then the model will select either A or B ' +
+                    'as the next token (using temperature).'
+            },
+            {
+                label: 'Top-k',
+                name: 'topK',
+                type: 'number',
+                step: 1,
+                optional: true,
+                additionalParams: true,
+                description:
+                    'Top-k changes how the model selects tokens for output.\n' +
+                    'A top-k of 1 means the selected token is the most probable among ' +
+                    'all tokens in the model vocabulary (also called greedy decoding), ' +
+                    'while a top-k of 3 means that the next token is selected from ' +
+                    'among the 3 most probable tokens (using temperature).'
+            }
+            /*
+            {
```

Comment:
can we remove commented unused code?

Reply:
Yes it is unused at the moment, but, OTOH, it is available for use in langchain.js (https://github.com/hwchase17/langchainjs/blob/main/langchain/src/chat_models/googlepalm.ts#L132) : It's just that Flowise doesn't make use of that available feature (also available in the VertexAI models, but unused).

I'm happy to remove the code - but put it in as a placeholder in case it inspired someone...  My guess is that langchain itself won't make use of this prior examples[] data until it appears in the OpenAI API.

Reply:
In the interests of getting this into 'main', I've removed the commented out code as first suggested.  The extra fields (and the associate UI) might be worth having if we're convinced that langchain would make good use of them (i.e. more efficiently implement few-shot prompts on the PaLM models), but my guess is the default option for the backend is to just add as previous context.  So, really, there's currently little to be gained from added these fields at the moment.

Reply:
that make sense, I'm good with that. we can always add in that if there is a need in the future